target/linux/octeon/base-files/lib/upgrade/platform.sh

```diff
@@ -1,123 +1,125 @@
+		mv "$kernel_file" "$kernel_file.previous"
+		mv "$kernel_checksum_file" "$kernel_checksum_file.previous"
 	fi
 
-	echo "flashing rootfs to ${rootfs}"
-	tar xf $tar_file $board_dir/root -O | dd of="${rootfs}" bs=4096
+	echo "Copying kernel to $boot_directory/$kernel_file"
+	tar xf "$sysupgrade_file" "$sysupgrade_directory/kernel" -O > "$kernel_file"
+	md5sum "$kernel_file" | cut -f1 -d " " > "$kernel_checksum_file"
+
+	echo "Flashing rootfs to $root_device"
+	tar xf "$sysupgrade_file" "$sysupgrade_directory/root" -O | dd of="$root_device" bs=4096
 
 	sync
-	umount /boot
+	umount "$boot_directory"
+	return 0
 }
 
 platform_do_upgrade() {
-	local tar_file="$1"
-	local board=$(board_name)
-	local rootfs="$(platform_get_rootfs)"
-	local kernel=
+	local sysupgrade_file="$1"
+	local kernel_file="vmlinux.64"
+	local boot_device
+	local root_device
 
-	[ -b "${rootfs}" ] || return 1
-	case "$board" in
-	er)
-		kernel=mmcblk0p1
-		;;
-	erlite)
-		kernel=sda1
-		;;
-	itus,shield-router)
-		kernel=ItusrouterImage
-		;;
-	*)
-		return 1
+	case "$(board_name)" in
+		er)
+			boot_device="/dev/mmcblk0p1"
+			root_device="/dev/mmcblk0p2"
+			;;
+		erlite)
+			boot_device="/dev/sda1"
+			root_device="/dev/sda2"
+			;;
+		itus,shield-router)
+			kernel_file="ItusrouterImage"
+			boot_device="/dev/mmcblk1p1"
+			root_device="/dev/mmcblk1p2"
+			;;
+		*)
+			return 1
 	esac
 
-	platform_do_flash $tar_file $board $kernel $rootfs
+	if [ -n "$boot_device" ] && \
+		[ -b "$boot_device" ] && \
+		[ -n "$root_device" ] && \
+		[ -b "$root_device" ] ; then
+		platform_do_flash "$sysupgrade_file" "$kernel_file" "$boot_device" "$root_device"
+	fi
 
 	return 0
 }
 
 platform_check_image() {
-	local board=$(board_name)
-	local tar_file="$1"
+	local sysupgrade_file="$1"
+	# find a sysupgrade directory by looking inside sysupgrade file
+	local sysupgrade_directory
+	sysupgrade_directory=$(tar tf "$sysupgrade_file" | awk -F '/' '/^sysupgrade/ {print $1; exit}')
+	[ -n "$sysupgrade_directory" ] || return 1
 
-	local board_dir=$(tar tf "$tar_file" | grep -m 1 '^sysupgrade-.*/$')
-	board_dir=${board_dir%/}
-	[ -n "$board_dir" ] || return 1
-
-	case "$board" in
+	case "$(board_name)" in
 	er | \
 	erlite | \
 	itus,shield-router)
-		local kernel_length=$(tar xf $tar_file $board_dir/kernel -O | wc -c 2> /dev/null)
-		local rootfs_length=$(tar xf $tar_file $board_dir/root -O | wc -c 2> /dev/null)
-		[ "$kernel_length" = 0 -o "$rootfs_length" = 0 ] && {
+		local kernel_length
+		local rootfs_length
+		kernel_length=$(tar xf "$sysupgrade_file" "$sysupgrade_directory/kernel" -O | wc -c 2> /dev/null)
+		rootfs_length=$(tar xf "$sysupgrade_file" "$sysupgrade_directory/root" -O | wc -c 2> /dev/null)
+		[ "$kernel_length" = 0 ] || [ "$rootfs_length" = 0 ] && {
 			echo "The upgrade image is corrupt."
 			return 1
 		}
 		return 0
 		;;
 	esac
 
-	echo "Sysupgrade is not yet supported on $board."
+	echo "Sysupgrade is not yet supported on $(board_name)."
```

Comment:
It's a good thing that you are motivated to improve style and clean things up.

However, please accept that we will only accept changes that represent an improvement, and not just change stuff. Many things here just do it differently, but not better. E.g. replacing $board by $board_name or separating declaration and assignment as for the kernel_length above.

Please just don't do that, it will essentially just make tracking any changes more complicated (as well as review).

As a general guideline, always think about what _is better_ after the change. If you cannot find something convincing to put into the commit message, the change is probably not needed and can be dropped.

Reply:
Same goes for renaming established variable names btw; "kernel" and "rootfs" are used in multiple scripts across the tree, so using different names here actually does not make it better, but worse.

Reply:
This `$board` changes is hardly improvement, previous state was better since it reused the variable, now it needs to spawn subshell to get the content every other time.